### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by RitaDee**
In this PR, I use models to insert data:

- Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- Open rails console for your project.
- Create at least one user by running the following code:
    ```ruby
       first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
       second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
    ```
- Create at least 4 posts written by one of the users you created by running the following code:
    ```ruby
       first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
    ```
- Create at least 6 posts comments for one of the posts you created by running the following code:
    ```ruby
       Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
    ```
- Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.

### Create custom methods
Your models should include some additional methods.
-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, foreign_key: :author_id, class_name: 'User', counter_cache: :posts_counter
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def update_user_post_counter
+    author.increment!(:postscounter)
+  end
```
Comment (codepantha) [inline app/models/post.rb:12]: - [x] [OPTIONAL] You have done a great job with this method, but we can make it even cooler! To do this, you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the #update_user_post_counter method automatically every time a new post is been saved to the database 😎 Then make the update_user_post_counter method private.

**See below 👇**

```ruby
after_save :update_user_post_counter

private

def update_user_post_counter
  author.increment!(:postscounter)
end
```

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class Comment < ApplicationRecord
-  belongs_to :user
-  belongs_to :post
+  belongs_to :author, foreign_key: :author_id, class_name: 'User'
+  belongs_to :post, class_name: 'Post', counter_cache: :comments_counter
+
+  def update_post_comment_counter
+    post.increment!(:comments_counter)
+  end
```
Comment (codepantha) [inline app/models/comment.rb:11]: - [x] [OPTIONAL] Please use the `after_save` rails callback here to update the post's comments counter just like I suggested for `posts`

File: app/models/like.rb [hunk 3]
```diff
@@ -1,4 +1,8 @@
 class Like < ApplicationRecord
-  belongs_to :user
-  belongs_to :post
+  belongs_to :author, foreign_key: :author_id, class_name: 'User'
+  belongs_to :post, counter_cache: :likescounter
+
+  def update_post_like_counter
+    post.increment!(:likescounter)
+  end
```
Comment (codepantha) [inline app/models/like.rb:11]: - [x] [OPTIONAL] Please use the `after_save` rails callback here to update the post's likes counter just like I suggested for `posts` and `comments`

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢  ✔️ 

## Hi @RitaDee 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (RitaDee) [reply app/models/post.rb:12]: Thank you @codepantha
I will explore that.

File: app/models/comment.rb (hunk 2 above)
Comment (RitaDee) [reply app/models/comment.rb:11]: Thanks. So, cool

**Pull request merged**